src/extractor/graph_compressor.cpp

```diff
@@ -217,6 +240,14 @@ void GraphCompressor::Compress(const std::unordered_set<NodeID> &barrier_nodes,
                 graph.GetEdgeData(forward_e1).duration += forward_duration2;
                 graph.GetEdgeData(reverse_e1).duration += reverse_duration2;
 
+                if (node_weight_penalty && node_duration_penalty)
+                {
+                    graph.GetEdgeData(forward_e1).weight += *node_weight_penalty;
+                    graph.GetEdgeData(reverse_e1).weight += *node_weight_penalty;
+                    graph.GetEdgeData(forward_e1).duration += *node_duration_penalty;
```

Comment:
From here additional duration penalties will be included into speed annotations.